templates/java.lang.String.h

```diff
@@ -2,6 +2,7 @@ String(const char* str);
 ~String();
```

Comment:
Just noticed, that we String class needs a custom move-constructor as well. The default implementation is bogus.